crates/core/src/lib.rs

```diff
@@ -97,21 +97,21 @@ impl Default for Config {
         // knobs for each of these settings just yet and instead they're
         // generally set to defaults. Environment-variable-based fallbacks are
         // supported though as an escape valve for if this is a problem.
-        //
-        // NB: much of this will change in Wasmtime 13 as the settings are
-        // different. Ping @alexcrichton for assistance in updating this if
-        // needed (and delete this comment after the 13 update).
         let mut pooling_config = PoolingAllocationConfig::default();
         pooling_config
-            .instance_count(env("SPIN_WASMTIME_INSTANCE_COUNT", 1_000))
-            .instance_size(env("SPIN_WASMTIME_INSTANCE_SIZE", (10 * MB) as u32) as usize)
-            .instance_tables(env("SPIN_WASMTIME_INSTANCE_TABLES", 2))
-            .instance_table_elements(env("SPIN_WASMTIME_INSTANCE_TABLE_ELEMENTS", 30_000))
-            .instance_memories(env("SPIN_WASMTIME_INSTANCE_MEMORIES", 1))
+            .total_component_instances(env("SPIN_WASMTIME_INSTANCE_COUNT", 1_000))
+            .max_component_instance_size(
+                env("SPIN_WASMTIME_INSTANCE_SIZE", (10 * MB) as u32) as usize
```

Comment:
Since this is carried over from my original changes, some documentation for this could be:

> This number accounts for internal data structures that Wasmtime allocates for each instance. Instance allocation is proportional to the number of "things" in a wasm module like functions, globals, memories, etc. Instance allocations are relatively small and are largely inconsequential compared to other runtime state, but a number needs to be chosen here so a relatively large threshold of 10MB is arbitrarily chosen. It should be unlikely that any reasonably-sized module hits this limit.

Reply:
Sorry I actually completely misread this before and thought it was `SPIN_WASMTIME_ISNTANCE_SIZE * 10MB`.